### DISCUSSION TITLE
Chrome autofill values on non-lazy fields are immediately cleared out

### DISCUSSION THREAD

**Discussion by divspace** [livewire/livewire#3059] [7. Bugs (archived)]
### Description

The fix for #2340 is what introduced this bug, because in Livewire 2.3.17 this bug does not exist.

When you autofill in Chrome, the values are immediately stripped out if they are called with `wire:model` or `wire:model.defer`.

Anything called with `wire:model.lazy` will keep the autofill values.

> In Safari, this will actually break because when you use autofill on a field that is lazy, the hydration process will set that value to `null` (so you can no longer type-hint), so it will fail validation as well even though there's a value "set." Once you fix the type-hinting by removing it or allowing for `null` too, auto-complete works. This doesn't seem to happen in Chrome though, I can only do this in Safari autofill with lazy fields that are type-hinted without accounting for `null` even though they are set to empty strings on mount.

### Exact steps to reproduce

Add three text fields to a form, one with `wire:model`, one with `wire:model.defer`, and one with `model:wire.lazy` and make sure they are fillable by the browser (so `name`, `email`, `zip_code`, etc.). Open the page and try to autofill the values, you'll see them get filled then stripped.

**Context**
- Livewire version: > 2.3.17
- Laravel version: 8.31.1
- Browser: Chrome, Safari

I'm currently tracking down what actually fixes this (besides staying on `2.3.17`).

**Comment by divspace** [comment]
This file on line 70 is the issue:

    js/node_initializer.js

Here's the difference:

```bash
@@ -70,7 +70,7 @@
             || directive.modifiers.includes('lazy') ? 'change' : 'input'

         // If it's a text input and not .lazy, debounce, otherwise fire immediately.
-        let handler = debounceIf(hasDebounceModifier || (DOM.isTextInput(el) && !isLazy), e => {
+        let handler = debounceIf(hasDebounceModifier || (DOM.isTextInput(el) && (!isLazy || !el.wasRecentlyAutofilled)), e => {
             let model = directive.value
             let el = e.target
```

When you revert this line back, autofill works again on both browsers, and you can change the type-hint to just `string` instead of `?string` to avoid the PHP error in Safari.

**Reply by leitom** [reply]
@calebporzio Updated livewire in a project today and got this issue again. The fix above solves the problem.

**Reply by calebporzio** [reply]
Hey there, this area of the codebase hasn't been changed in 7 months. When was the last time you updated and which was the exact  version that broke things for you? Thanks!

**Comment by leitom** [comment]
Thank you for finding a solution to this @divspace It was a "mind boggling" issue :)

**Comment by weeg** [comment]
Was this bug fixed? Because in version 2.5.5 it seems this still exists.

**Reply by weeg** [reply]
And I have tried it in iOS and Mac OS Safari.

**Reply by rasmuscnielsen** [reply]
Yeah I can definitely second this. Using wire:model.lazy in MacOS Safari + iOS safari just clears the input :-/

**Furthermore** I found a potential issue / edge case in Chrome:

Given a form has BOTH .defer + .lazy inputs, only the .lazy inputs will be persisted when auto completing the entire form in Chrome. All .defer inputs will be cleared. 

In my case I was needing to populate a select-input only when address-fields changed and simply have contact fields defered. But this triggered the above mentioned issue. 

Tested in Livewire 2.5.5 and Chrome Version 91.0.4472.114.

**Comment by scottzirkel** [comment]
We have had to set all of our forms to defer due to this issue. It's a pretty major one, and honestly there's no point in even running the forms through Livewire at this point. I had posted #3029 and my coworker posted #3027 if that helps anyone.

**Reply by EriksonEnterprises** [reply]
This is still a problem to this day :/ setting to defer made it work for the most part.

I was hoping to somehow have live validation that would validate autofilled data too, looks like that's not an option lol
